src/engine/Value.ts

```diff
@@ -92,6 +90,44 @@ export abstract class Value<
   }
 }
 
+export class BoolValue extends Value<boolean> {
+  constructor(val: boolean) {
+    super(val || false);
+  }
+  public get isTruthy() {
+    return Boolean(this.value);
+  }
+  public get valueType() {
+    return ValueType.Int;
+  }
+
+  public Cast(newType: ValueType): Value<any> {
+    if (this.value === null) return throwNullException("Value.value");
```

Comment:
I kept this in to stay consistent with the other casts, but:

1. It's not in the original source
2. It's not actually needed in this case because `null` gets evaluated to false

But I think we probably don't really want to consider null and false the same in this case.

Reply:
I honestly can't remember when this check was introduced and why!

Probably because if `this.value` is `null` in C#, the casts (at the Ink level, not at the language) would throw a null pointer exception while in Javascript, they would just go through silently.

It's probably trying to match the behaviour of the original runtime.